Libplanet/PoS/Control/Bond.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using Libplanet.Action;
+using Libplanet.Assets;
+
+namespace Libplanet.PoS
+{
+    internal static class Bond
+    {
+        internal static (IAccountStateDelta, FungibleAssetValue) Execute(
+            IAccountStateDelta states,
+            FungibleAssetValue consensusToken,
+            Address validatorAddress,
+            Address delegationAddress)
+        {
+            // TODO: Failure condition
+            // 1. Validator does not exist
+            // 2. Exchange rate is invalid(validator has no tokens but there are outstanding shares)
+            // 3. Amount is less than the minimum amount
+            // 4. Delegator does not have sufficient consensus token (fail or apply maximum)
+            if (!consensusToken.Currency.Equals(Asset.ConsensusToken))
+            {
+                throw new ArgumentException();
```

Comment:
IMO the exception was a bit more descriptive. like `InvalidBondCurrencyException(actual, expected)`.

Reply:
Applied on commit 8045f68c000f18f2e9b8608ae7c2b33ccfc96cd0.